### ISSUE TITLE
Update of the page can be sluggish

### ISSUE THREAD

**Issue by LupinIII** [OsaSoft/youtube-better-subscriptions#34]
When toggling the checkbox or marking a video the youtube page felt like it needed quite the pause to think and update.

I have made a fork/branch where I changed how the DOM is traversed and the video ID is read:
[https://github.com/LupinIII/youtube-better-subscriptions/tree/get-thumbnail-container-once](url)
[https://github.com/LupinIII/youtube-better-subscriptions/tree/regexp-for-video-id](url)
This made the addon more responsive for me.

(I have made a few other branches where I fix something I had a problem with, or check the master branch in my fork where I combined all of it)

**Comment by OsaSoft** [comment]
Hey, thanks for the input. Ill gladly take a look at the improvements youve made and consider merging them, if youd be ok with that

**Comment by LupinIII** [comment]
Sure, that's why I made a public fork. Pick what you think is useful. You might want to make the changes in your source instead of a real merge though, since I may not have used the same coding style.
Btw. I just added a new commit to my master branch (that's what I run locally) that trims the storage to a certain number of entries by removing the oldest ones. Just in case storage space is limited on a system.
